extensions/blocks/map/map.php

```diff
@@ -25,7 +25,83 @@
 function jetpack_map_block_load_assets( $attr, $content ) {
 	$api_key = Jetpack_Options::get_option( 'mapbox_api_key' );
 
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		global $wp, $map_block_counter;
+		if ( ! $map_block_counter ) {
+			$map_block_counter = 0;
+		}
+		$iframe_url = home_url( $wp->request ) . '?map-block-counter=' . $map_block_counter;
+
+		$placeholder = preg_replace( '/(?<=<div\s)/', 'placeholder ', $content );
+
+		// @todo Is intrinsic size right? Is content_width the right dimensions?
+		return sprintf(
+			'<amp-iframe src="%s" width="%d" height="%d" layout="intrinsic" allowfullscreen sandbox="allow-scripts">%s</amp-iframe>',
+			esc_url( $iframe_url ),
+			Jetpack::get_content_width(),
+			Jetpack::get_content_width(),
+			$placeholder
+		);
+	}
+
 	Jetpack_Gutenberg::load_assets_as_required( 'map' );
 
 	return preg_replace( '/<div /', '<div data-api-key="' . esc_attr( $api_key ) . '" ', $content, 1 );
 }
+
+/**
+ * Render a page containing only a single Map block.
+ */
+function jetpack_map_block_render_single_block_page() {
+	$map_block_counter_as_string = filter_input( INPUT_GET, 'map-block-counter', FILTER_SANITIZE_STRING );
```

Comment:
```suggestion
	$map_block_counter_as_string = (int) filter_input( INPUT_GET, 'map-block-counter', FILTER_SANITIZE_NUMBER_INT );
```

Reply:
Won't this return 0 if `map-block-counter` is not present?

Reply:
Here's an alternate approach with a 1-based counter to work around the problem:

https://github.com/Automattic/jetpack/pull/13405/commits/4a1a2dfa425e2d0fff9c5944e4fe37d15a00c936